src/Akka.Persistence.Sql/Journal/Dao/BaseByteArrayJournalDao.cs

```diff
@@ -112,22 +112,37 @@ public async Task<IImmutableList<Exception>> AsyncWriteMessages(
 
         public async Task Delete(string persistenceId, long maxSequenceNr)
         {
+            long maxMarkedDeletion;
+
+            // no need to use transaction
+            await using (var connection = ConnectionFactory.GetConnection())
+            {
+                maxMarkedDeletion = await MaxMarkedForDeletionMaxPersistenceIdQuery(connection, persistenceId, maxSequenceNr).FirstOrDefaultAsync(ShutdownToken);
+            }
+
+            if (maxMarkedDeletion is 0)
+                return;
+            
             await ConnectionFactory.ExecuteWithTransactionAsync(
                 WriteIsolationLevel,
                 ShutdownToken,
                 async (connection, token) =>
                 {
-                    await connection
-                        .GetTable<JournalRow>()
+                    var journalTable = connection.GetTable<JournalRow>();
+                    await journalTable
                         .Where(
                             r =>
                                 r.PersistenceId == persistenceId &&
-                                r.SequenceNumber <= maxSequenceNr)
+                                r.SequenceNumber == maxMarkedDeletion)
                         .Set(r => r.Deleted, true)
                         .UpdateAsync(token);
 
-                    var maxMarkedDeletion =
-                        await MaxMarkedForDeletionMaxPersistenceIdQuery(connection, persistenceId).FirstOrDefaultAsync(token);
+                    await journalTable
+                        .Where(
+                            r =>
+                                r.PersistenceId == persistenceId &&
+                                r.SequenceNumber < maxMarkedDeletion)
+                        .DeleteAsync(token);
```

Comment:
Simplify the deletion query, just delete anything that is less than the highest affected row.